zone/spells.cpp

```diff
@@ -1040,87 +1040,162 @@ bool Mob::CheckFizzle(uint16 spell_id)
-		case 3:
-			specialize = specialize * 1.3;
-			break;
+		// CALCULATE EFFECTIVE SPECIALIZATION SKILL VALUE
+		float specialize_skill_value = GetSpecializeSkillValue(spell_id);
+		switch (GetAA(aaSpellCastingMastery)) {
+			case 1:
+				specialize_skill_value = specialize_skill_value * 1.05;
+				break;
+			case 2:
+				specialize_skill_value = specialize_skill_value * 1.15;
+				break;
+			case 3:
+				specialize_skill_value = specialize_skill_value * 1.3;
+				break;
 		}
-		if(((specialize/6.0f) + 15.0f) < zone->random.Real(0, 100)) {
-			specialize *= SPECIALIZE_FIZZLE / 200.0f;
-		} else {
-			specialize = 0.0f;
+
+		float specialize_reduction = (specialize_skill_value > 50) ? (specialize_skill_value - 50) / 10 : 0.0f;
+
+		// CALCULATE EFFECTIVE CASTING STAT VALUE
+		float prime_stat_reduction = 0.0f;
+
+		if (GetCasterClass() == 'W') {
+			prime_stat_reduction = (GetWIS() - 75) / 10.0;
+		} else if (GetCasterClass() == 'I') {
+			prime_stat_reduction = (GetINT() - 75) / 10.0;
 		}
-	}
 
-	// == 0 --> on par
-	// > 0 --> skill is lower, higher chance of fizzle
-	// < 0 --> skill is better, lower chance of fizzle
-	// the max that diff can be is +- 235
-	float diff = par_skill + static_cast<float>(spells[spell_id].base_difficulty) - act_skill;
+		// BARDS ARE SPECIAL - they add both CHA and DEX mods to get casting rates similar to full casters without spec skill
+		if (GetClass() == Class::Bard) {
+			prime_stat_reduction = (GetCHA() - 75 + GetDEX() - 75) / 10.0;
+		}
 
-	// if you have high int/wis you fizzle less, you fizzle more if you are stupid
-	if(GetClass() == Class::Bard)
-	{
-		diff -= (GetCHA() - 110) / 20.0;
-	}
-	else if (GetCasterClass() == 'W')
-	{
-		diff -= (GetWIS() - 125) / 20.0;
-	}
-	else if (GetCasterClass() == 'I')
-	{
-		diff -= (GetINT() - 125) / 20.0;
-	}
+		// GET SPELL-SPECIFIC FIZZLE CHANCE (note that specialization is only used to reduce the Fizzle_adjust!)
+		float spell_fizzle_adjust = static_cast<float>(spells[spell_id].base_difficulty);
+		spell_fizzle_adjust = (spell_fizzle_adjust - specialize_reduction > 0) ? spell_fizzle_adjust - specialize_reduction : 0.0f;
+
+		// CALCULATE FINAL FIZZLE CHANCE
+		float fizzle_chance = spell_difficulty + spell_fizzle_adjust - caster_skill - prime_stat_reduction;
+
+		if (fizzle_chance > 95.0f) {
+			fizzle_chance = 95.0f;
+		} else if (fizzle_chance < 2.0f) {
+			fizzle_chance = 2.0f;
+		}
+
+		float fizzle_roll = zone->random.Real(0, 100);
 
-	// base fizzlechance is lets say 5%, we can make it lower for AA skills or whatever
-	float basefizzle = 10;
-	float fizzlechance = basefizzle - specialize + diff / 5.0;
+		LogSpells("Check Fizzle [{}]  spell: [{}]  fizzle_chance: [{}]  roll: [{}]", GetName(), spell_id, fizzle_chance, fizzle_roll);
 
-	// always at least 1% chance to fail or 5% to succeed
-	fizzlechance = fizzlechance < 1 ? 1 : (fizzlechance > 95 ? 95 : fizzlechance);
+		if (fizzle_roll > fizzle_chance) {
+			return true;
+		}
+
+		return false;
+	} else {
```

Comment:
We can remove the entire `else {` enclosure - if we're using legacy spell code we return out of this function anyways and if we're not we fall through to this logic regardless of an else